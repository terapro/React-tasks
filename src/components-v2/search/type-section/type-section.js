import React from 'react';
import PropTypes from 'prop-types';
import {Label} from 'src/components-v2/common/label/label.js';
import {Button} from 'src/components-v2/common/button/button.js';
import {parameters} from 'src/data/constants.js';

export const TypeSection = ({activeSearchType, searchTypeClick}) =>
 (
    <div className={'search-type-section'}>
      <Label content={'search by'} labelUC labelWhite />
      {
        parameters.searchTypes.map((item) => (
            <Button btnRed={item.value === activeSearchType}
                    btnGrey={item.value !== activeSearchType}
                    key = {item.value}
                    id = {item.value}
                    content = {item.name}
                    btnClick={searchTypeClick}
                    btnInline
                    btnUC
            />
          )
        )
      }
    </div>
  );

TypeSection.propTypes = {
    searchTypeClick: PropTypes.func,
    activeSearchType: PropTypes.string
};

TypeSection.defaultProps = {
    searchTypeClick: ()=> {}
};
