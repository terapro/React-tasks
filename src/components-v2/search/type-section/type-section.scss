.search-type-section {
  display: flex;
}

.search-type-section > div:nth-child(2n){
  margin: 0 $elementPadding;
}


