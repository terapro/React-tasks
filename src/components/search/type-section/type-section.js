import React from 'react';
import PropTypes from 'prop-types';

import {Label} from 'src/components/common/label/label.js';
import {Button} from 'src/components/common/button/button.js'

const TypeSection = (props) => {
  const {searchTypes, searchTypeCallback} = props;
  return (
    <div className={'search-type-section'}>
      <Label content={'search by'} labelUC labelWhite />
      {
        searchTypes.list.map((item) => (
            <Button btnRed={item === searchTypes.active}
                    btnGrey={item !== searchTypes.active}
                    key = {item}
                    id = {item}
                    content = {item}
                    btnClick={searchTypeCallback}
                    btnInline
                    btnUC
            />
          )
        )
      }
    </div>
  );

};


TypeSection.propTypes = {
  searchTypes: PropTypes.shape({
    list: PropTypes.arrayOf(PropTypes.string),
    active: PropTypes.string
  }).isRequired,
  searchTypeCallback: PropTypes.func
};
TypeSection.defaultProps = {
  searchTypeCallback: ()=> {}
};

export {TypeSection};