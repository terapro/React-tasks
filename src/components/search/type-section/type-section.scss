.search-type-section {
  color: #FFF;
  font-weight: bold;
  font-size: 16px;
  text-transform: uppercase;
}