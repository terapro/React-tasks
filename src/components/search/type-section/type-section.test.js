import React from 'react';
import {TypeSection} from 'src/components/search/type-section/type-section';
import {shallow} from 'enzyme';

const mockSearchTypes = {
  list: ['title', 'genres'],
  active: 'title'
};

const mockFunc = jest.fn();

describe('<TypeSection />', () => {
  describe('Render', () => {
    it('should render the component', () => {
      const wrapper = shallow(<TypeSection searchTypes={mockSearchTypes}
                                           searchTypeClick={mockFunc}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});