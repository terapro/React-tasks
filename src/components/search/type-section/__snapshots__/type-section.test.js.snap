// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeSection /> Render should render the component 1`] = `
<div
  className="search-type-section"
>
  <Label
    content="search by"
    labelBold={false}
    labelUC={true}
    labelWhite={true}
  />
  <Button
    btnBig={false}
    btnBold={false}
    btnClick={[MockFunction]}
    btnGrey={false}
    btnInline={true}
    btnRed={true}
    btnSm={false}
    btnTransp={false}
    btnTranspRed={false}
    btnUC={true}
    btnWhite={false}
    content="title"
    id=""
    key="title"
  />
  <Button
    btnBig={false}
    btnBold={false}
    btnClick={[MockFunction]}
    btnGrey={true}
    btnInline={true}
    btnRed={false}
    btnSm={false}
    btnTransp={false}
    btnTranspRed={false}
    btnUC={true}
    btnWhite={false}
    content="genres"
    id=""
    key="genres"
  />
</div>
`;
